client/pages/filespage/thing-existing.js

```diff
@@ -239,9 +239,11 @@ export class ExistingThing extends React.Component {
         }
         className = className.trim();
 
+        let encodedLink = this.props.file.link.split('/').map(val => encodeURIComponent(val)).join('/')
```

Comment:
This part of the code can be called quite extensively (eg: when a directory has thousands of files), I'd rather avoid loop as much as possible around there

Reply:
See my latest commit - instead of iterating over the entire split path, I'm now performing an encode then replacing the `%2F`, which is something you're doing elsewhere in the codebase as well.